lib/SILOptimizer/Transforms/AccessEnforcementOpts.cpp

```diff
@@ -150,21 +160,30 @@ using DenseAccessSet = SmallVector<BeginAccessInst *, 4>;
 /// or at the end_access instruction that is associated with the
 /// begin_access.
 ///
-/// Forward data flow computes `blockOutAccess` for each block. At a control
-/// flow merge, this analysis forms an intersection of reachable accesses on
-/// each path. Only visited predecessors are merged (unvisited paths
-/// optimistically assume reachability). Before a block is visited, it has no
-/// map entry in blockOutAccess. Blocks are processed in RPO order, and a single
+/// Forward data flow computes `regionToStorageMap` for each region's blocks.
+/// Regions are processed bottom-up in RPO order.
+/// At a block's control flow merge, this analysis forms an intersection of
+/// reachable accesses on each path inside the region.
+/// Only visited predecessors are merged (unvisited paths optimistically
+/// assume reachability). Before a block is visited, it has no map entry in
+/// regionToStorageMap. Blocks are processed in RPO order, and a single
 /// begin_access dominates all associated end_access instructions. Consequently,
-/// when a block is first visited, blockOutAccess contains the maximal
+/// when a block is first visited, its storage accesses contains the maximal
 /// reachability set. Further iteration would only reduce this set.
 ///
-/// The only result of this analysis is the seenNestedConflict flags in
-/// AccessInfo. Since reducing a reachability set cannot further detect
-/// conflicts, there is no need to iterate to a reachability fix point.
-class AccessConflictAnalysis {
+/// The only results of this analysis are:
+//// 1) The seenNestedConflict flags in AccessInfo.
+///     Since reducing a reachability set cannot further detect
+///     conflicts, there is no need to iterate to a reachability fix point.
+///  2) A deterministic order map of out-of-scope instructions that we can
+///  merge.
+///     The way we construct this map guarantees the accesses within it are
+///     mergeable.
+class RegionStorageAnalysis {
```

Comment:
This is analyzing an entire function at once, and doing more than just summarizing the accesses within a region, I don't think it should be called "RegionStorageAnalysis". AccessConflictAnalysis seemed fitting.